server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/model/Keywords.scala

```diff
@@ -0,0 +1,134 @@
+    case Left(errorMessage: String) => {
+      println(errorMessage)
+      None
+    }
+    case Right(value: Keyword) => Some(value)
+  }
+}
+
+trait KeywordsValidator {
+      def validate(keywords: Set[Keyword]): Boolean
+}
+
+object KeywordsValidator {
+      val THROW_ON_IMAP_NON_EXPOSED_KEYWORDS: KeywordsValidator = (keywords: Set[Keyword]) => {
+        if (keywords.exists(KeywordFilter.FILTER_IMAP_NON_EXPOSED_KEYWORDS)) {
+          throw new IllegalArgumentException("Does not allow to update 'Deleted' or 'Recent' flag")
+        }
+        true
+      }
+
+      val IGNORE_NON_EXPOSED_IMAP_KEYWORDS: KeywordsValidator = (_: Set[Keyword]) => { None[Boolean] }
+    }
+
+object KeywordFilter {
+      val FILTER_IMAP_NON_EXPOSED_KEYWORDS: Keyword => Boolean = (keyword: Keyword) => keyword.isExposedImapKeyword()
+      val KEEP_ALL: Keyword => Boolean = (_: Keyword) => true
+}
+
+object KeywordsFactory {
+    val STRICT_KEYWORDS_FACTORY = new KeywordsFactory(KeywordsValidator.THROW_ON_IMAP_NON_EXPOSED_KEYWORDS, KeywordFilter.KEEP_ALL, ToKeyword.STRICT)
+    val LENIENT_KEYWORDS_FACTORY = new KeywordsFactory(KeywordsValidator.IGNORE_NON_EXPOSED_IMAP_KEYWORDS, KeywordFilter.FILTER_IMAP_NON_EXPOSED_KEYWORDS, ToKeyword.LENIENT)
+
+    private[model] trait KeywordFilter extends IterableOnce[Keyword] {}
+}
+
+object Keywords {
+  val DEFAULT_VALUE: Set[Keywords] = Set()
+}
+
+final case class Keywords(keywords: Set[Keyword]) {
+
+  def combiner(firstBuilder: FlagsBuilder, secondBuilder: FlagsBuilder): FlagsBuilder = firstBuilder.add(secondBuilder.build)
+  def accumulator(accumulator: FlagsBuilder, keyword: Keyword): FlagsBuilder = accumulator.add(keyword.asFlags())
+
+  /**
+   * This Keywords factory will filter out invalid keywords (containing inavlid character, being too long or being non
+   * exposed IMAP flags)
+   */
+  def strictFactory: KeywordsFactory = STRICT_KEYWORDS_FACTORY
+
+  /**
+   * This Keywords factory will throw upon invalid keywords (containing inavlid character, being too long or being non
+   * exposed IMAP flags)
+   *
+   * @throws IllegalArgumentException
+   */
+  def lenientFactory: KeywordsFactory = LENIENT_KEYWORDS_FACTORY
+
+  def asFlags(): Flags = keywords.asJava.stream().reduce(FlagsBuilder.builder, this.accumulator, this.combiner).build
+
+  def asFlagsWithRecentAndDeletedFrom(originFlags: Flags): Flags = {
+    val flags = asFlags
+    if (originFlags.contains(Flags.Flag.DELETED)) flags.add(Flags.Flag.DELETED)
+    if (originFlags.contains(Flags.Flag.RECENT)) flags.add(Flags.Flag.RECENT)
+    flags
+  }
+
+  def asMap: Map[Keyword, Boolean] = keywords.map((_, true)).toMap
+
+  def getKeywords: Set[Keyword] = keywords
+
+  def contains(keyword: Keyword): Boolean = keywords.contains(keyword)
+}
+
+class KeywordsFactory(val validator: KeywordsValidator, val filter: Keyword => Boolean, val toKeyword: ToKeyword) {
+    def fromSet(setKeywords: Set[Keyword]): Keywords = {
```

Comment:
indent?

Reply:
Let's return a `Try[Keyword]` as this method contains a sneaky throw